src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -204,6 +204,11 @@ public static BinaryData FromObjectAsJson<T>(T jsonSerializable, JsonSerializerO
         /// <seealso cref="ToObjectFromJson{String}" />
         public override unsafe string ToString()
         {
+            if (_bytes.Length == 0)
+            {
+                return string.Empty;
+            }
+
             ReadOnlySpan<byte> span = _bytes.Span;
```

Comment:
```suggestion
            ReadOnlySpan<byte> span = _bytes.Span;

            if (span.IsEmpty)
            {
                return string.Empty;
            }

```
It is tiny bit better to check the `span` since the value is cached in a local.